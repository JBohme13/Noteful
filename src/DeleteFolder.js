import React, { Component } from 'react'
import NotefulContext from './NotefulContext'
import ValidationError from './ValidationError'
import './DeleteFolder.css'

export default class DeleteFolder extends Component {
    static contextType = NotefulContext;

    render() {
      const value = this.context;
      const deleteFolderError = value.deleteFolderError
        return(
          <section id='delete-folder-container'>
            <h2>Delete folder</h2>
            <form id='delete-folder-form'>
                <label htmlFor='select-folder-for-delete'>Select folder</label>
                <select 
                    type='radio'
                    name='select-folder-for-delete'
                    id='select-folder-for-delete'
                    required='required'
                    aria-required='true'
                  >
                      {
                          value.folders.map((folder, i) => {
                            return(
                              <option 
                                key={i}
                                value={folder.id}
                              >
                                  {folder.name}
                              </option>
                            )
                          })
                      }
                  </select>
                  <br />
                  <ValidationError message={deleteFolderError}/>
                  <button
                      id='delete-button' 
                      type='submit'
                      onClick={event => value.deleteFolder(event, document.getElementById('select-folder-for-delete').value)}
                  >
                      Delete
                  </button>
            </form>
            <button
                id='clear-button' 
                onClick={event => value.handleClearButton(event)}
            >
                Clear
            </button>
          </section>
        )
    }
}