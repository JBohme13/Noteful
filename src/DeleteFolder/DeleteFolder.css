#delete-folder-form {
    display: flex;
    flex-direction: column;
}

#select-folder-for-delete {
    order: 1;
    flex: 2;
    height: 30px;
    font-size: 14px;
    width: 55%;
    margin: 5px;
    color: darkslategrey;
}

#delete-button {
    order: 2;
    height: 22px;
    margin: 5px;
    border: 1px solid grey;
    border-radius: 2px;
    box-shadow: 1px 1px;
    font-size: 14px;
    color: darkslategrey;
}

#delete-button {
    box-shadow: -1px -1px;
}

#clear-button {
    position: relative;
    left: 82px;
    top: -32px;
    height: 22px;
    margin: 5px;
    border: 1px solid grey;
    border-radius: 2px;
    box-shadow: 1px 1px;
    font-size: 14px;
    color: darkslategrey;
}

#clear-button {
    box-shadow: -1px -1px;
}

label {
    color: darkslategrey;
    font-size: 16px;
}

h2 {
    color: darkslategrey;
    font-weight: normal;
}