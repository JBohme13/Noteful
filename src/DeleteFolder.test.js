import React from 'react';
import ReactDOM from 'react-dom';
import DeleteFolder from './DeleteFolder';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('DeleteFolder component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<DeleteFolder />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<DeleteFolder />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  /*it('Deletes folder', () => {
    const wrapper = shallow(<DeleteFolder />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })*/
});