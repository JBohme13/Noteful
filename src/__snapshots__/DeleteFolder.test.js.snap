// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteFolder component renders the UI as expected 1`] = `
<div
  id="delete-folder-container"
>
  <h2>
    Delete folder
  </h2>
  <form
    id="delete-folder-form"
  >
    <label
      htmlFor="select-folder-for-delete"
    >
      Select folder
    </label>
    <select
      id="select-folder-for-delete"
      name="select-folder-for-delete"
      required="required"
      type="radio"
    />
    <button
      id="delete-button"
      onClick={[Function]}
      type="submit"
    >
      Delete
    </button>
  </form>
  <button
    id="clear-button"
    onClick={[Function]}
  >
    Clear
  </button>
</div>
`;
